factory/src/main/kotlin/io/qalipsis/core/factory/meters/CampaignMeterRegistryImpl.kt

```diff
@@ -244,58 +283,55 @@ internal class CampaignMeterRegistryImpl(
     private fun <T> gauge(meterId: Meter.Id, stateObject: T, valueFunction: ToDoubleFunction<T>): Gauge {
         return meters.computeIfAbsent(meterId) {
             GaugeImpl(
-                MicrometerGauge.builder(meterId.meterName, stateObject, valueFunction)
-                    .tags(meterId.tags.forMicrometer())
-                    .register(meterRegistry),
                 meterId,
                 meterReporter
-            )
+            ) { valueFunction.applyAsDouble(stateObject) }
         } as Gauge
     }
 
     override fun summary(name: String, vararg tags: String): DistributionSummary {
-        return summary(asId(name, tagsAsList(tags)))
+        return summary(asId(name, tagsAsList(tags), MeterType.DISTRIBUTION_SUMMARY))
     }
 
-    override fun summary(name: String, tags: Iterable<Tag>): DistributionSummary {
-        return summary(asId(name, tags))
+    override fun summary(name: String, tags: Map<String, String>): DistributionSummary {
+        return summary(asId(name, tags, MeterType.DISTRIBUTION_SUMMARY))
     }
 
     override fun summary(
         scenarioName: ScenarioName,
         stepName: StepName,
         name: String,
-        tags: Map<String, String>
+        tags: Map<String, String>,
     ): DistributionSummary {
         return summary(
             Meter.Id(
                 campaignKey = currentCampaignKey,
                 scenarioName = scenarioName,
                 stepName = stepName,
                 meterName = name,
-                tags = tags
+                tags = tags + additionalTags,
+                type = MeterType.DISTRIBUTION_SUMMARY
             )
         )
     }
 
     private fun summary(meterId: Meter.Id): DistributionSummary {
         return meters.computeIfAbsent(meterId) {
             DistributionSummaryImpl(
-                MicrometerDistributionSummary.builder(meterId.meterName)
-                    .tags(meterId.tags.forMicrometer())
-                    .register(meterRegistry),
                 meterId,
-                meterReporter
+                meterReporter,
+                percentiles,
```

Comment:
I think that we should let the possibility to overwrite the percentiles and histogram count when building the meter. There should be passed as optional arguments of the creation of the summary.